interview-questions.md

```diff
@@ -8,62 +8,66 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is iteration?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: Iteration is the condition of running a function repeatedly until it meets a condition that you have specified.
 
+  Researched answer: Iteration is the process where a set of instructions or statements is executed repeatedly for a specified number of time or until a condition is met.
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: The predefined parameters of map takes the whole array and returns the same length of the original array after doing something to everything in the array. Not sure which are optional.
 
-  Researched answer:
+  Researched answer: Map is a collection of elements where each element is stored as a Key, value pair. Map object can hold both objects and primitive values as either key or value. When we iterate over the map object it returns the key,value pair in the same order as inserted.
 
 
 
 3. What is object destructuring?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: Object destructuring is the process of using the object brackets to get the values out that are in the object to use more easily.
 
+  Researched answer: Destructuring is a JavaScript expression that allows us to extract data from arrays, objects, and maps and set them into new, distinct variables. Destructuring allows us to extract multiple properties, or items, from an array​ at a time.
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: There is no real difference they are pretty much the same. A method is a built in function within Javascript.
 
-  Researched answer:
+  Researched answer: A function can be called directly by its name.	A method consists of a code that can be called by the name of its object and its method name using dot notation or square bracket notation..
+  A function can pass the data that is operated and may return the data. 	The method operates the data contained in a Class.
+  Data passed to a function is explicit.	A method implicitly passes the object on which it was called.
+  A function lives on its own.	A method is a function associated with an object property.
```

Comment:
Awesome research response. I did find exact matches to one of these sentences on google but I'll let it slide I like this response so much.